cgi-bin/LJ/User/Administration.pm

```diff
@@ -329,6 +329,22 @@ sub unban_user_multi {
 
 
 ########################################################################
+### Selective Screening functions
+
+# return if $target's comments will automatically be screened in $u's journal
+sub has_autoscreen {
+    my ( $u, $target ) = @_;
+
+    my $uid = LJ::want_userid( $u );
+    my $jid = LJ::want_userid( $target );
+    return 1 unless $uid && $jid;
```

Comment:
You are entirely correct. I'd copied the code from has_banned, which neatly avoids the problem by only ever being called in places where the logic is such that anon posters never hit the function call. has_autoscreen is in *almost* the same situation - six of the seven function calls are wrapped in similar logic. The one that isn't is, of course, the one that actually determines whether the comment should be screened at all (the others are all logic about whether to display the "this comment will be screened" message at the top of the replybox).

It's still good to have that early return for anons, I think, to avoid having to make a check_rel call just to find out that no, we're not autoscreening anons. So I've flipped the return value to 0.